packages/api/src/car.js

```diff
@@ -116,52 +116,56 @@ export async function handleCarUpload (request, env, ctx, car, uploadType = 'Car
     structure = 'Complete'
   }
 
-  const [{ cid, pins }, backupKey] = await Promise.all([
-    addToCluster(car, env),
-    backup(car, rootCid, user._id, env, structure)
-  ])
+  const bucketKey = await uploadToBucket(env.s3Client, env.s3BucketName, car, rootCid, user._id, structure)
 
   const xName = headers.get('x-name')
   let name = xName && decodeURIComponent(xName)
   if (!name || typeof name !== 'string') {
     name = `Upload at ${new Date().toISOString()}`
   }
 
-  const normalizedCid = normalizeCid(cid)
+  const normalizedCid = normalizeCid(rootCid.toString())
   await env.db.createUpload({
     user: user._id,
     authKey: authToken?._id,
     contentCid: normalizedCid,
-    sourceCid: cid,
+    sourceCid: rootCid.toString(),
     name,
     type: uploadType,
-    backupUrls: backupKey
-      ? [`https://${env.s3BucketName}.s3.${env.s3BucketRegion}.amazonaws.com/${backupKey}`]
-      : [],
-    pins,
+    backupUrls: [`https://${env.s3BucketName}.s3.${env.s3BucketRegion}.amazonaws.com/${bucketKey}`],
+    pins: [],
     dagSize
   })
 
   /** @type {(() => Promise<any>)[]} */
-  const tasks = []
+  const tasks = [async () => {
+    let pins = []
+    try {
+      pins = await addToCluster(car, env)
+    } catch (err) {
+      env.log.warn(new Error('failed to upload to cluster', { cause: err }))
+      pins = await pinToCluster(rootCid.toString(), env)
```

Comment:
💡 we could pin first then add for maximum chance of actually getting the blocks on to cluster at some point... in the edge case where addToCluster runs out of function invokation time.